apps/website/src/middleware.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  type NextFetchEvent,
+  type NextRequest,
+  NextResponse,
+} from "next/server";
+import { env } from "@/env/index.mjs";
+import { callEndpoint } from "@/server/utils/queue";
+import type { TrackClickSchema } from "@/pages/api/short-links/track-click";
+
+export async function middleware(req: NextRequest, event: NextFetchEvent) {
+  const res = await fetch(env.NEXT_PUBLIC_BASE_URL + "/api/short-links", {
+    method: "POST",
+    body: JSON.stringify({
+      slug: req.nextUrl.pathname.replace("/l/", "").trim(),
+    }),
+  });
+  const shortLink = await res.json();
+
+  if (shortLink) {
+    event.waitUntil(
+      callEndpoint<TrackClickSchema>("/api/short-links/track-click", {
+        id: shortLink.id,
+      }),
+    );
+    return NextResponse.redirect(shortLink.link);
+  }
+
+  // If no match is found, redirect the user to the 404 page
+  return NextResponse.redirect(env.NEXT_PUBLIC_BASE_URL + "/404");
```

Comment:
Out of curiosity, what happens if we return nothing here -- does Next.js serve a 404 without a redirect happening?

Reply:
It serves the Next.js standart 404 page
![image](https://github.com/alveusgg/alveusgg/assets/90050350/b58d2ef4-c847-4a7a-ad9d-434e8b89ef08)

Reply:
Maybe my local env is bugged. If I open a normal page with an invalid url I also get the next.js 404 page

Reply:
But `.next` should work